GeosCore/convection_mod.F90

```diff
@@ -616,24 +652,29 @@ SUBROUTINE DO_CLOUD_CONVECTION( Input_Opt,  &
     !-----------------------------------------------------------------
     ! Compute PDOWN and BMASS
     !-----------------------------------------------------------------
-    DO K = 1, NLAY
-
-       ! PDOWN is the convective precipitation leaving each
-       ! box [cm3 H2O/cm2 air/s]. This accounts for the
-       ! contribution from both liquid & ice precip.
-       ! PFLCU and PFICU are converted from kg/m2/s to m3/m2/s
-       ! using water and ice densities, respectively.
-       ! m3/m2/s becomes cm3/cm2/s using a factor of 100.
-       PDOWN(K) = ( ( PFLCU(K) / 1000e+0_fp ) &
-                +   ( PFICU(K) /  917e+0_fp ) ) * 100e+0_fp
-
-       ! BMASS is the dry air mass per unit area for the grid box
-       ! bounded by level K and K+1 [kg/m2]
-       ! BMASS is equivalent to deltaP (dry) * 100 / g
-       ! This is done to keep BMASS in the same units as CMFMC * SDT
-       BMASS(K) = DELP_DRY(K) * G0_100
+    ! PDOWN is the convective precipitation leaving each
+    ! box [cm3 H2O/cm2 air/s]. This accounts for the
+    ! contribution from both liquid & ice precip.
+    ! PFLCU and PFICU are converted from kg/m2/s to m3/m2/s
+    ! using water and ice densities, respectively.
+    ! m3/m2/s becomes cm3/cm2/s using a factor of 100.
+    IF ( Is_GF_Conv == .FALSE. ) THEN
+       PDOWN(:) = ( ( PFLCU(:) / 1000e+0_fp ) &
+                +   ( PFICU(:) /  917e+0_fp ) ) * 100e+0_fp
+    ELSE
+       PDOWN(NLAY) = 0
+       DO K = NLAY-1, 1, -1
+          PDOWN(K) = PDOWN(K+1) + DQRCU_MET(K+1) & 
```

Comment:
Shouldn't this be: PDOWN(K) = PDOWN(K+1) + DQRCU_MET(**K**) * DELP(**K**) * G0_100 * 100e+0_fp

Reply:
I did this to align with the pre-existing configuration:
https://github.com/geoschem/geos-chem/blob/bef56c605e018eecbd91646a51ce82c7cd77f56a/GeosCore/convection_mod.F90#L569-L575
But if we look at where PDOWN is used, it seems to imply that PDOWN(K) to be the flux at the bottom of level K. So I think you are right, and we need to fix both RAS and GF's PDOWN. 

Also, I have a question about:
https://github.com/geoschem/geos-chem/blob/bef56c605e018eecbd91646a51ce82c7cd77f56a/GeosCore/convection_mod.F90#L1101-L1107
Why is essential to have re-evaporation to have wash-out? Couldn't it be washout event without re-evaporation?

Reply:
You are right. Re-evaporation should not be a condition for washout, but I guess there will always be re-evaporation when the precip falls through drier air below the cloud base. Could you remove the re-evaporation condition and see how much difference it makes? 
Let's fix PDOWN in RAS too. Thanks!

Reply:
Hi Viral, please check the latest commit.

Reply:
Hi @yuanjianz, this looks great! One thing I could not tell for certain is whether we go through the entire routine even if there is no convection taking place in a column at the time step, or if there is an early exit somewhere if, say, the convective mass flux is zero throughout the column.